index.bs

```diff
@@ -712,44 +731,47 @@ Processing model {#processing}
+created by running [=create brands=] with [=user agent/significant version=].
 
-<h4 id="create-ua-list-section">Create brands</h4>
+Every {{WindowOrWorkerGlobalScope}} object has an associated
+<dfn for="WindowOrWorkerGlobalScope">brands frozen array</dfn>, which is a
+<code><a interface>FrozenArray</a>&lt;<a dictionary>NavigatorUABrandVersion</a>></code>. It is
+initially the result of [=create a frozen array|creating a frozen array=] from the [=user agent=]'s
+[=brands=].
 
-When asked to run the <dfn>create brands</dfn> algorithm, the [=user agent=] MUST run the following steps:
-1. Let |list| be a [=/list=].
+Additionally, every {{WindowOrWorkerGlobalScope}} object has an associated
+<dfn for="WindowOrWorkerGlobalScope">version list frozen array</dfn>, which is a
+<code><a interface>FrozenArray</a>&lt;<a dictionary>NavigatorUABrandVersion</a>></code>. It is
+the result of [=create a frozen array|creating a frozen array=] from running [=create brands=] with
+[=user agent/full version=].
 
-2. Collect pairs of [=user agent/brand=] and [=user agent/significant version=] which represent the [=user agent=] or
-    [=equivalence classes=].
+<h4 algorithm="to create brands" id="create-ua-list-section">Create brands</h4>
 
-3. For each pair:
+When asked to <dfn>create brands</dfn> with |version type|, run the following steps:
 
-    1. Let |dict| be a new {{NavigatorUABrandVersion}} dictionary,
-        with [=user agent/brand=] as {{NavigatorUABrandVersion/brand}} and [=user agent/significant version=] as
+1. Let |list| be a [=/list=].
+1. Collect pairs of [=user agent/brand=] and versions of type |version type| which represent the
+    [=user agent=] or [=equivalence classes=].
+1. For each pair:
+    1. Let |dict| be a new {{NavigatorUABrandVersion}} dictionary, with [=user agent/brand=] as
+        {{NavigatorUABrandVersion/brand}} and a string of type |version type| as
         {{NavigatorUABrandVersion/version}}.
-
-    2. Append |dict| to |list|.
-
-4.  The [=user agent=] SHOULD execute the following steps:
-
-    1.  [=list/Append=] one additional [=list/item=] to |list| containing a {{NavigatorUABrandVersion}} dictionary,
-        initialized with <a lt="create an arbitrary brand">arbitrary brand</a> and <a lt="create an arbitrary version">
-        arbitrary version</a> combinations.
-
-    2.  Randomize the order of the [=list/items=] in |list|.
+    1. Append |dict| to |list|.
+1.  The [=user agent=] SHOULD execute the following steps:
+    1. [=list/Append=] one additional [=list/item=] to |list| containing a
+        {{NavigatorUABrandVersion}} dictionary, initialized with
+        <a lt="create an arbitrary brand">arbitrary brand</a> as {{NavigatorUABrandVersion/brand}}
+        and the result of <a lt="create an arbitrary version">creating an arbitrary version</a> with
+        {{NavigatorUABrandVersion/version}}.
+    1.  Randomize the order of the [=list/items=] in |list|.
```

Comment:
Should the order in the two lists be the same or could they be randomized to be in different orders? If it should be the same, randomization should be done elsewhere and passed along here. If they can be different, this is fine.

Reply:
I prefer random, to keep things intentional (and not encourage folks to assume orders are stable anywhere).